extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReaderImpl.java

```diff
@@ -199,15 +172,21 @@ public SeekableChannelsProvider getChannelsProvider() {
     }
 
     @NotNull
-    private Dictionary readDictionary(long dictionaryPageOffset, SeekableChannelContext channelContext) {
+    private Dictionary readDictionary(SeekableChannelContext channelContext) {
+        // This will cause io.deephaven.parquet.table.TestParquetTools.testNoDictionaryOffset to fail
+        // Should this be exposed as an option in ParquetInstructions instead?
+        // Ideally, we should would prefer to trust columnChunk by default
+        // if (!columnChunk.hasDictionaryPage()) {
+        // return NULL_DICTIONARY;
+        // }
         // Use the context object provided by the caller, or create (and close) a new one
         try (
                 final ContextHolder holder = SeekableChannelContext.ensureContext(channelsProvider, channelContext);
-                final SeekableByteChannel ch =
-                        channelsProvider.getReadChannel(holder.get(), getURI()).position(dictionaryPageOffset)) {
+                final SeekableByteChannel ch = channelsProvider.getReadChannel(holder.get(), getURI())) {
+            ch.position(columnChunk.getStartingPos());
```

Comment:
This has a new check, that the dictionary offset is before the first data page offset. Do we think that's just about rejecting malformed metadata that claims a dictionary page that cannot exist?